src/lib/supports_pixelated_image.js

```diff
@@ -15,7 +15,8 @@ function supportsPixelatedImage() {
     if(_supportsPixelated !== null) { // only run the feature detection once
         return _supportsPixelated;
     }
-    if(Lib.isIE()) {
+    if(Lib.isIE() || Lib.isSafari() || Lib.isIOS()) {
+        // NB. Safari passes the test below but the final rendering is not pixelated
```

Comment:
We should possibly disable this feature when `window.navigator.userAgent` is empty.
Also to make debugging easier we may rewrite this function something like: 
```js
function supportsPixelatedImage() {
    if(_supportsPixelated !== null) { // only run the feature detection once
        return _supportsPixelated;
    }

    _supportsPixelated = false;

    if(!(
        !window.navigator.userAgent ||
        Lib.isSafari() ||
        Lib.isIOS() ||
        Lib.isIE()
    )) {
        var declarations = Array.from(constants.CSS_DECLARATIONS).reverse();

        var supports = (window.CSS && window.CSS.supports) || window.supportsCSS;
        if(typeof supports === 'function') {
            _supportsPixelated = declarations.some(function(d) {
                return supports.apply(null, d);
            });
        } else {
            var image3 = Drawing.tester.append('image')
                .attr('style', constants.STYLE);

            var cStyles = window.getComputedStyle(image3.node());
            var imageRendering = cStyles.imageRendering;

            _supportsPixelated = declarations.some(function(d) {
                var value = d[1];
                return (
                    imageRendering === value ||
                    imageRendering === value.toLowerCase()
                );
            });

            image3.remove();
        }
    }

    return _supportsPixelated;
}
```